src/lib/flowCommand.ts

```diff
@@ -21,9 +21,16 @@ const COMMANDS = {
   XPUB: 'xpub',
 };
 
+type FlowType = {
+  kind?: 'takeLockup' | 'btc' | 'xpub';
+  lock?: 'linear' | 'conditional';
+  from?: string;
+  utx?: string;
+} | null;
```

Comment:
Based and typespilled.

But this doesn't quite capture what we want/expect here. The keys are mutually exclusive based on the `kind`. I think you want something like:

```suggestion
type FlowType =
  | { kind: 'takeLockup', lock: 'linear' | 'conditional', from: string }
  | { kind: 'btc', utx: string }
  | { kind: 'xpub' }
  | null;
```

Reply:
Good point, this communicates the actual use case.

Like the `BridgeWallet` type, this type was created in the midst of a debugging session, but ultimately I didn't up using it :) So, will remove it from the PR ✂️